tap_outbrain/__init__.py

```diff
@@ -344,37 +344,20 @@ def sync_links(state, access_token, account_id, campaign_id):
     logger.info('Done syncing links for campaign {}.'.format(campaign_id))
 
 
-def do_sync(args):
+def discover():
+    catalog = {"streams": STREAMS}
+    return singer.Catalog.from_dict(catalog)
+
+def do_sync(config, catalog: singer.Catalog):
     global DEFAULT_START_DATE
     state = DEFAULT_STATE
 
-    with open(args.config) as config_file:
-        config = json.load(config_file)
-    missing_keys = []
-    if 'username' not in config:
-        missing_keys.append('username')
-    else:
-        username = config['username']
-
-    if 'password' not in config:
-        missing_keys.append('password')
-    else:
-        password = config['password']
-
-    if 'account_id' not in config:
-        missing_keys.append('account_id')
-    else:
-        account_id = config['account_id']
-
-    if 'start_date' not in config:
-        missing_keys.append('start_date')
-    else:
-        # only want the date
-        DEFAULT_START_DATE = config['start_date'][:10]
+    username = config['username']
+    password = config['password']
+    account_id = config['account_id']
 
-    if len(missing_keys) > 0:
-        logger.fatal("Missing {}.".format(", ".join(missing_keys)))
-        raise RuntimeError
```

Comment:
This is now handled by [`singer.utils.check_config`](https://github.com/dbt-labs/tap-outbrain/blob/7ea59804ef2a7ce10a205787c3c3a51dccbb3b78/tap_outbrain/__init__.py#L382).